props/websubmit/src/main.rs

```diff
@@ -85,16 +93,98 @@ impl DeletionProp {
     }
 }
 
+/// Storing data in the database must be associated to a user. This is
+/// necessary for e.g. the deletion to work.
+pub struct ScopedStorageProp {
+    cx: Arc<PolicyContext>,
+}
+
+impl ScopedStorageProp {
+    pub fn new(cx: Arc<PolicyContext>) -> Self {
+        ScopedStorageProp { cx }
+    }
+
+    pub fn check(self) {
+        for c_id in self.cx.desc().controllers.keys() {
+            let scopes = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|node| self.cx.has_marker(marker!(scopes), *node))
+                .collect::<Vec<_>>();
+            let stores = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|node| self.cx.has_marker(marker!(stores), *node))
+                .collect::<Vec<_>>();
+            let mut sensitives = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|node| self.cx.has_marker(marker!(sensitive), *node));
+
+            let controller_valid = sensitives.all(|sens| {
+                stores.iter().all(|&store| {
+                    // sensitive flows to store implies some scope flows to store callsite
+                    !(self
```

Comment:
Since this condition is happening inside a sequence of `all`'s it would be nicer for debugging to do the `assert_error!` here with a message that includes which data isn't stored properly and the store it flows to since we have that info here.

Reply:
Done